src/core/core.cpp

```diff
@@ -327,6 +327,20 @@ wayfire_view wf::compositor_core_t::get_cursor_focus_view()
     return view ? view->self() : nullptr;
 }
 
+wf::surface_interface_t *wf::compositor_core_impl_t::get_surface_at(wf::pointf_t point)
+{
+    wf::pointf_t local = {0.0, 0.0};
+    return input->input_surface_at(point, local);
+}
+
+wayfire_view wf::compositor_core_t::get_view_at(wf::pointf_t point)
+{
+    auto surface = get_surface_at(point);
+    auto view = dynamic_cast<wf::view_interface_t*> (surface);
```

Comment:
this is wrong. You should try to cast `surface->get_main_surface()`, otherwise, this will fail if you are over a subsurface.